### ISSUE TITLE
`tmux` sometimes replays environment variables set by previous activations

### ISSUE THREAD

**Issue by limeytexan** [flox/flox#1848]
**Describe the bug:**

I'm going to describe this in some detail because while the problem happens to me every few weeks or so, I have not spent the time to find a specific reproducer.

**The problem:**

This is how the problem presents itself:

- started fresh terminal
- did `flox activate -r limeytexan/default`
- the `$_{add,del}_env` env variables were correctly set to correspond to PID `$$`
- invoked `tmux`
- the subshell of `tmux` then had entirely different `$_{add,del}_env` variables referring to a completely different PID(!)
    - those files didn't exist, so I didn't even get a `PATH` set

**The workaround:**

I hunted around everywhere for mention of this "other" PID and could not find mention of it anywhere. Figuring it had to be `tmux` setting the variable I then poked around in the source to no avail, set logging, etc. (oh how I wish for `strace` on Mac!!!) and in the end I found that this made the problem go away:

- ensure `tmux list-sessions` returns nothing
- run `rm -rf /tmp/tmux-*`

Funny thing is, the `/tmp/tmux-501` directory only contains a socket called "default":
```
flox [limeytexan/default (remote)] brantley@Michaels-MacBook-Pro myflox % find /tmp/tmux-501 -ls
 32022165      0 drwx------   3 brantley wheel          96 Jul  5 13:03 /tmp/tmux-501
 32022167      0 srw-rw----   1 brantley wheel           0 Jul  5 13:03 /tmp/tmux-501/default
flox [limeytexan/default (remote)] brantley@Michaels-MacBook-Pro myflox % 
```

I don't understand what's going on yet, but I wanted to document the workaround FTR.

**Steps to reproduce:**

N/A

**flox Version (run `flox --version` if possible):**

1.2.2

**`uname -a` output:**

```
Michaels-MBP:myflox brantley$ tmux -V
tmux 3.4
Michaels-MBP:myflox brantley$ uname -a
Darwin Michaels-MBP.localdomain 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:17:33 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6031 arm64 arm Darwin
```

**Comment by limeytexan** [comment]
Got frustrated with this today and started researching again, and found this article: https://stackoverflow.com/questions/50506708/how-to-start-two-tmux-sessions-with-different-environments

TL;DR: start a new tmux session with `tmux -L $$` (or any random value) to create a new server socket with a completely distinct "global" configuration and set of env variables.

Also useful: `tmux showenv -g` shows all the things that tmux will set for you. I haven't yet figured out a way to clear out that list, but I tend to think this is feature is distinctly at odds with what we're doing by way of flox activations in subshells. 🤔

**Comment by limeytexan** [comment]
Ah ... progress
```
flox [limeytexan/default (remote)] brantley@Michaels-MacBook-Pro ~ % tmux showenv -g | cut -d= -f1 | xargs -n 1 tmux set-environment -g -u
flox [limeytexan/default (remote)] brantley@Michaels-MacBook-Pro ~ % tmux showenv -g                                                      
flox [limeytexan/default (remote)] brantley@Michaels-MacBook-Pro ~ % 
```
